fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneOnlineIndexingTest.java

```diff
@@ -145,6 +153,115 @@ void luceneOnlineIndexingTestSimple() {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            assertTrue(recordStore.isIndexWriteOnly(index));
+            recordStore.saveRecord(createComplexDocument(1623L, ENGINEER_JOKE, 1, startTime));
+            context.commit();
+        }
+
+        // validate a new partition meta has been created
+        List<LucenePartitionInfoProto.LucenePartitionInfo> partitionInfos = luceneIndexTestValidator.getPartitionMeta(index, Tuple.from(1));
+        assertEquals(1, partitionInfos.size());
+        assertEquals(1, partitionInfos.get(0).getCount());
+        assertEquals(Tuple.from(startTime).addAll(Tuple.from(1, 1623L)), LucenePartitioner.getPartitionKey(partitionInfos.get(0)));
+        assertEquals(Tuple.from(startTime).addAll(Tuple.from(1, 1623L)), LucenePartitioner.getToTuple(partitionInfos.get(0)));
+
+        // update another record; partition information exist, but the record isn't indexed yet
+        // this will also succeed
+        try (final FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            assertTrue(recordStore.isIndexWriteOnly(index));
+            recordStore.saveRecord(createComplexDocument(1547L, ENGINEER_JOKE, 1, startTime + 1_000));
+            context.commit();
+        }
+
+        // validate partition meta has been correctly updated
+        partitionInfos = luceneIndexTestValidator.getPartitionMeta(index, Tuple.from(1));
+        assertEquals(1, partitionInfos.size());
+        assertEquals(2, partitionInfos.get(0).getCount());
+        assertEquals(Tuple.from(startTime).addAll(Tuple.from(1, 1623L)), LucenePartitioner.getPartitionKey(partitionInfos.get(0)));
+        assertEquals(Tuple.from(startTime + 1_000).addAll(Tuple.from(1, 1547L)), LucenePartitioner.getToTuple(partitionInfos.get(0)));
+
+        // finish the indexing process
+        try (final FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .build()) {
+                assertTrue(recordStore.isIndexWriteOnly(index));
+                indexBuilder.buildIndex(true);
+            }
+        }
+
+        // validate index is sane
+        luceneIndexTestValidator.validate(
+                index,
+                Map.of(
+                        Tuple.from(1), Map.of(
+                                Tuple.from(1, 899L), Tuple.from(startTime + 3_000),
+                                Tuple.from(1, 2222L), Tuple.from(startTime + 2_000),
+                                Tuple.from(1, 1547L), Tuple.from(startTime + 1_000),
+                                Tuple.from(1, 1623L), Tuple.from(startTime)
+                        )
+                ),
+                2,
+                "*:*");
```

Comment:
This test won't catch if either flow results in not-updating the index, it just makes sure that all the values are there. It would be good to do searches that verify that the index has the updated values.